community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/DefaultRelationshipTraversalCursor.java

```diff
@@ -507,6 +510,12 @@ private void reset()
     @Override
     protected void collectAddedTxStateSnapshot()
     {
+        if ( filterState == FilterState.NOT_INITIALIZED )
```

Comment:
Nope. Here we ensure that the filter state is initialized so that we get the correct tx-state relationships, and therefore `filterStore` does not matter.